ibm_mq/README.md

```diff
@@ -2,47 +2,170 @@
+
+Update your LD_LIBRARY_PATH to include the location of the libraries. For example:
+
+```
+export LD_LIBRARY_PATH=/opt/mqm/lib64:/opt/mqm/lib:$LD_LIBRARY_PATH
+```
+
+*Note*: Agent 6 uses Upstart or systemd to orchestrate the datadog-agent service. Environment variables may need to be added to the service configuration files at the default locations of /etc/init/datadog-agent.conf (Upstart) or /lib/systemd/system/datadog-agent.service (systemd). See documentation on Upstart or systemd for more information on how to configure these settings.
+
+Here's an example of the configuration that's used for Systemd:
+
+```yaml
+[Unit]
+Description="Datadog Agent"
+After=network.target
+Wants=datadog-agent-trace.service datadog-agent-process.service
+StartLimitIntervalSec=10
+StartLimitBurst=5
+
+[Service]
+Type=simple
+PIDFile=/opt/datadog-agent/run/agent.pid
+Environment="LD_LIBRARY_PATH=/opt/mqm/lib64:/opt/mqm/lib:$LD_LIBRARY_PATH"
+User=dd-agent
+Restart=on-failure
+ExecStart=/opt/datadog-agent/bin/agent/agent run -p /opt/datadog-agent/run/agent.pid
+
+[Install]
+WantedBy=multi-user.target
+```
+
+Here's an example of the upstart config:
+
+```
+description "Datadog Agent"
+
+start on started networking
+stop on runlevel [!2345]
+
+respawn
+respawn limit 10 5
+normal exit 0
+
+# Logging to console from the Agent is disabled since the Agent already logs using file or
+# syslog depending on its configuration. We make Upstart log what the process still outputs in order
+# to log panics/crashes to /var/log/upstart/datadog-agent.log
+console log
+env DD_LOG_TO_CONSOLE=false
+env LD_LIBRARY_PATH=/opt/mqm/lib64:/opt/mqm/lib:$LD_LIBRARY_PATH
+
+setuid dd-agent
+
+script
+  exec /opt/datadog-agent/bin/agent/agent start -p /opt/datadog-agent/run/agent.pid
+end script
+
+post-stop script
+  rm -f /opt/datadog-agent/run/agent.pid
+end script
+```
+
+Alternatively, if you are using Linux, after the MQ Client is installed ensure the runtime linker can find the libraries. For example, using ldconfig:
+
+```
+# Put the library location in an ld configuration file.
+
+sudo sh -c "echo /opt/mqm/lib64 > /etc/ld.so.conf.d/mqm64.conf"
+sudo sh -c "echo /opt/mqm/lib > /etc/ld.so.conf.d/mqm.conf"
+
+# Update the bindings.
+
+sudo ldconfig
+```
+
+#### Permissions
+
+There are a number of ways to set up permissions in IBM MQ. Depending on how your setup works, you will want to create a datadog user with read only permissions.
+
 
 ### Configuration
 
 1. Edit the `ibm_mq.d/conf.yaml` file, in the `conf.d/` folder at the root of your
-   Agent's configuration directory to start collecting your ibm_mq performance data.
-   See the [sample ibm_mq.d/conf.yaml][3] for all available configuration options.
+   Agent's configuration directory to start collecting your IBM MQ performance data.
+   See the [sample ibm_mq.d/conf.yaml][4] for all available configuration options.
+
+2. [Restart the Agent][5]
+
+#### Metrics
```

Comment:
```suggestion
#### Metric Collection
```

Reply:
This is the title we generate -- are we sure it's not tied to some kind of automation?

Reply:
No, this part isn't. The part under `Data Collected` is though.

I suggested naming to match what we've done for other logs-enabled integrations, eg: [Apache](https://github.com/DataDog/integrations-core/tree/master/apache), which Nils suggested earlier.